ksp_plugin/plugin.cpp

```diff
@@ -937,18 +941,75 @@ void Plugin::ComputeAndRenderNodes(
                Vector<double, Navigation>({0, 0, 1}),
                ascending_trajectory,
                descending_trajectory);
-  ascending = renderer_->RenderPlottingTrajectoryInWorld(
-                  current_time_,
-                  ascending_trajectory.Begin(),
-                  ascending_trajectory.End(),
-                  sun_world_position,
-                  PlanetariumRotation());
-  descending = renderer_->RenderPlottingTrajectoryInWorld(
-                   current_time_,
-                   descending_trajectory.Begin(),
-                   descending_trajectory.End(),
-                   sun_world_position,
-                   PlanetariumRotation());
+
+  std::optional<DiscreteTrajectory<Navigation>> filtered_ascending_trajectory;
+  std::optional<DiscreteTrajectory<Navigation>> filtered_descending_trajectory;
+
+  auto const cast_plotting_frame = dynamic_cast<
+      BodyCentredNonRotatingDynamicFrame<Barycentric, Navigation> const*>(
+      &*renderer_->GetPlottingFrame());
+  if (cast_plotting_frame != nullptr) {
+    auto const& plotting_frame = *cast_plotting_frame;
+    auto const centre = dynamic_cast_not_null<RotatingBody<Barycentric> const*>(
```

Comment:
Do `*dynamic_cast_not_null` and you'll avoid all the `->` below.

Reply:
I `dynamic_cast_not_null<OblateBody<Barycentric> const*>(centre)` below, and that becomes `&dynamic_cast<OblateBody<Barycentric> const&>(centre)`, which, being passed as a `not_null` parameter, induces a gratuitous nullness check.
Do we want that?

Reply:
Ah, I missed that.